lib/hurley/client.rb

```diff
@@ -329,6 +328,20 @@ def inspect
       ]
     end
 
+    private
+
+    def set_receiver
+      statuses, receiver = @body_receiver
+
+      if @receiver != receiver && (!statuses || statuses.include?(@status_code))
+        if @receiver.respond_to?(:join)
```

Comment:
This demonstrates the issue raised in #28 that this isn't true streaming. The body is going to be read before the response is returned, all your doing is buffering it in the initial BodyReceiver then forwarding it once the status code is set.